src/core/mod.rs

```diff
@@ -70,46 +72,78 @@ fn null_or_mut_ptr<T>(s: &mut Vec<u8>) -> *mut T {
     }
 }
 
-impl BPF {
-    #[cfg(any(
-        feature = "v0_4_0",
-        feature = "v0_5_0",
-        feature = "v0_6_0",
-        feature = "v0_6_1",
-        feature = "v0_7_0",
-        feature = "v0_8_0",
-    ))]
-    /// `code` is a string containing C code. See https://github.com/iovisor/bcc for examples
-    pub fn new(code: &str) -> Result<BPF, BccError> {
-        let cs = CString::new(code)?;
-        let ptr = unsafe { bpf_module_create_c_from_string(cs.as_ptr(), 2, ptr::null_mut(), 0) };
-        if ptr.is_null() {
-            return Err(BccError::Compilation);
-        }
+#[cfg(any(
+    feature = "v0_4_0",
+    feature = "v0_5_0",
+    feature = "v0_6_0",
+    feature = "v0_6_1",
+    feature = "v0_7_0",
+    feature = "v0_8_0",
+))]
+/// `code` is a string containing C code. See https://github.com/iovisor/bcc for examples
+fn module_create(code: &str, cflags: Vec<&str>) -> Result<*mut c_void, BccError> {
+    let ptr = unsafe { bpf_module_create_c_from_string(cs.as_ptr(), 2, ptr::null_mut(), 0) };
+    if ptr.is_null() {
+        return Err(BccError::Compilation);
+    }
 
-        Ok(BPF {
-            p: AtomicPtr::new(ptr),
-            uprobes: HashSet::new(),
-            kprobes: HashSet::new(),
-            tracepoints: HashSet::new(),
-            raw_tracepoints: HashSet::new(),
-            perf_events: HashSet::new(),
-            perf_events_array: HashSet::new(),
-            perf_readers: Vec::new(),
-            sym_caches: HashMap::new(),
-        })
+    Ok(ptr)
+}
+
+// 0.9.0 changes the API for bpf_module_create_c_from_string()
+#[cfg(any(feature = "v0_9_0", feature = "v0_10_0",))]
+/// `code` is a string containing C code. See https://github.com/iovisor/bcc for examples
+fn module_create(code: &str, cflags: Vec<&str>) -> Result<*mut c_void, BccError> {
```

Comment:
Just realised I only implemented it for my featureset, will add the changes for the other features soon. Sorry about this! :/